### PR TITLE
added dish nutrition servlet, minor fixes to pom and dishanalysis

### PR DESCRIPTION
**Pull request by joeychin01**
Added dishNutritionServlet that takes a dishName and returns json nutrition information from Spoonacular https://spoonacular.com/food-api/docs#Guess-Nutrition-by-Dish-Name

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public class SpoonacularNutritionServlet extends HttpServlet {
+  static String spoonacularPrefix = "https://api.spoonacular.com/recipes/guessNutrition";
+  static String spoonacularAPIKey = "cd2269d31cb94065ad1e73ce292374a5";
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String title = request.getParameter("dishName");
+    Client client = ClientBuilder.newClient();
+    WebTarget target = client.target(spoonacularPrefix + "?title=" + title + "&apiKey=" + spoonacularAPIKey);
+    try {
+      String recipeListJSONString = target.request(MediaType.APPLICATION_JSON).get(String.class);
```
Comment (cocovsack) [inline src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java:43]: change this name to indicate nutritional content.

File: src/main/java/com/google/sps/servlets/DishAnalysisServlet.java [hunk 2]
```diff
@@ -73,7 +72,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       List<AnnotateImageResponse> responses = new_response.getResponsesList();
 
       // Initialize blocked catagories
-      Set<String> blockedCatagories = new HashSet<String>;
+      Set<String> blockedCatagories = new HashSet<String>();
       blockedCatagories = new HashSet<String>(Arrays.asList("Cuisine", "Dish", "Food", "Ingredient", "Salad", "Fried food"));
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/DishAnalysisServlet.java:76]: This should be a constant of ImmutableSet type.

File: src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import javax.ws.rs.core.MediaType;  
+import com.google.gson.Gson;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+/** Servlet to take in dish name and return bulk recipe infomation */
+@WebServlet("/dishNutrition")
+public class SpoonacularNutritionServlet extends HttpServlet {
+  static String spoonacularPrefix = "https://api.spoonacular.com/recipes/guessNutrition";
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java:35]: private static final String PREFIX = "...";
private static final String API_KEY = "...";

File: src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  static String spoonacularAPIKey = "cd2269d31cb94065ad1e73ce292374a5";
+  @Override
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java:37]: Add one line separator above and add javadoc.

File: src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String title = request.getParameter("dishName");
+    Client client = ClientBuilder.newClient();
+    WebTarget target = client.target(spoonacularPrefix + "?title=" + title + "&apiKey=" + spoonacularAPIKey);
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java:41]: Consider using String.format()

**Pull request merged**